props/saber_sabersense_buttons.h

```diff
@@ -0,0 +1,1017 @@
+EFFECT(battery);  // for EFFECT_BATTERY_LEVEL
+EFFECT(bmbegin);  // for Begin Battle Mode
+EFFECT(bmend);    // for End Battle Mode
+EFFECT(vmbegin);  // for Begin Volume Menu
+EFFECT(vmend);    // for End Volume Menu
+EFFECT(volup);    // for increse volume
+EFFECT(voldown);  // for decrease volume
+EFFECT(volmin);   // for minimum volume reached
+EFFECT(volmax);   // for maximum volume reached
+EFFECT(faston);   // for EFFECT_FAST_ON
+EFFECT(blstbgn);  // for Begin Multi-Blast
+EFFECT(blstend);  // for End Multi-Blast
+EFFECT(push);     // for Force Push gesture in Battle Mode
+EFFECT(quote);    // for playing quotes
+
+
+
+// The Saber class implements the basic states and actions
+// for the saber.
+class SabersenseButtons : public PROP_INHERIT_PREFIX PropBase {
+public:
+  SabersenseButtons() : PropBase() {}
+  const char* name() override { return "SabersenseButtons"; }
+
+  void Loop() override {
+    PropBase::Loop();
+    DetectTwist();
+    Vec3 mss = fusor.mss();
+    if (SaberBase::IsOn()) {
+      DetectSwing();
+      if (auto_lockup_on_ &&
+          !swinging_ &&
+          fusor.swing_speed() > 120 &&
+          millis() - clash_impact_millis_ > SABERSENSE_LOCKUP_DELAY &&
+          SaberBase::Lockup()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_lockup_on_ = false;
+      }
+      if (auto_melt_on_ &&
+          !swinging_ &&
+          fusor.swing_speed() > 60 &&
+          millis() - clash_impact_millis_ > SABERSENSE_LOCKUP_DELAY &&
+          SaberBase::Lockup()) {
+        SaberBase::DoEndLockup();
+        SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+        auto_melt_on_ = false;
+      }
+
+      // EVENT_PUSH
+      if (fabs(mss.x) < 3.0 &&
+          mss.y * mss.y + mss.z * mss.z > 70 &&
+          fusor.swing_speed() < 30 &&
+          fabs(fusor.gyro().x) < 10) {
+        if (millis() - push_begin_millis_ > SABERSENSE_FORCE_PUSH_LENGTH) {
+          Event(BUTTON_NONE, EVENT_PUSH);
+          push_begin_millis_ = millis();
+        }
+      } else {
+        push_begin_millis_ = millis();
+      }
+
+    } else {
+      // EVENT_SWING - Swing On gesture control to allow fine tuning of speed needed to ignite
+      if (millis() - saber_off_time_ < MOTION_TIMEOUT) {
+        SaberBase::RequestMotion();
+        if (swinging_ && fusor.swing_speed() < 90) {
+          swinging_ = false;
+        }
+        if (!swinging_ && fusor.swing_speed() > SABERSENSE_SWING_ON_SPEED) {
+          swinging_ = true;
+          Event(BUTTON_NONE, EVENT_SWING);
+        }
+      }
+      // EVENT_THRUST
+      if (mss.y * mss.y + mss.z * mss.z < 16.0 &&
+          mss.x > 14  &&
+          fusor.swing_speed() < 150) {
+        if (millis() - thrust_begin_millis_ > 15) {
+          Event(BUTTON_NONE, EVENT_THRUST);
+          thrust_begin_millis_ = millis();
+        }
+      } else {
+        thrust_begin_millis_ = millis();
+      }
+    }
+  }
+
+// Volume Menu
```

Comment:
indent comment two steps

Reply:
Hoping the formatting is now OK as I've removed all the tabs and and hopefully changed it to preferred convention.